tests/lib.rs

```diff
@@ -199,4 +200,33 @@ mod util {
             }
         }
     }
+
+    /// Without Right::Lookup, looking up paths in a directory is not allowed
+    #[test]
+    fn test_basic_dir_err() {
+        let tdir = tempdir().unwrap();
+        let dir = Directory::new(tdir.path()).unwrap();
+        let fname = "foo";
+        fs::File::create(tdir.path().join(fname)).unwrap();
+        let rights = RightsBuilder::new(Right::Read).finalize().unwrap();
+        rights.limit(&dir).unwrap();
+        match unsafe { fork() }.unwrap() {
+            ForkResult::Child => {
+                always_abort();
+                capsicum::enter().unwrap();
+                let e = dir.open_file(fname, 0, None).unwrap_err();
+                // The OS should return ENOTCAPABLE, but std::io::ErrorKind
+                // doesn't have a kind for that.
+                if matches!(e, CapErr::Invalid(_)) {
```

Comment:
nit: possibly a reasonable follow-up, but I believe `CapErr` doesn't provide a good way to access the underlying `io::Error` and we'll always get `CapErr::Invalid` for a failed open [here](https://github.com/dlrobertson/capsicum-rs/blob/master/src/util.rs#L69), so we're not really providing much useful info to the user in the failure case.

Reply:
Let's revisit this as part of #25 .